src/web.cpp

```diff
@@ -123,90 +211,200 @@ void handle_handlestatus()
+            ADD_STR(json, "garageDoorState", "Unknown");
         }
-        json.append(std::string("  \"garageDoorState\": \"") + doorState + "\",\n");
     }
-    if (all || argReq["lockstate"])
+    if (all || HAS_ARG("lockstate"))
     {
-        std::string lockState = "";
         switch (garage_door.current_lock)
         {
         case 0:
-            lockState = "Unsecured";
+            ADD_STR(json, "garageLockState", "Unsecured");
             break;
         case 1:
-            lockState = "Secured";
+            ADD_STR(json, "garageLockState", "Secured");
             break;
         case 2:
-            lockState = "Jammed";
+            ADD_STR(json, "garageLockState", "Jammed");
             break;
         default:
-            lockState = "Unknown";
+            ADD_STR(json, "garageLockState", "Unknown");
         }
-        json.append(std::string("  \"garageLockState\": \"") + lockState + "\",\n");
     }
-    if (all || argReq["lighton"])
-        json.append(std::string("  \"garageLightOn\": ") + (garage_door.light ? "true" : "false") + ",\n");
-    if (all || argReq["motion"])
-        json.append(std::string("  \"garageMotion\": ") + (garage_door.motion ? "true" : "false") + ",\n");
-    if (all || argReq["obstruction"])
-        json.append(std::string("  \"garageObstructed\": ") + (garage_door.obstructed ? "true" : "false") + ",\n");
+    if (all || HAS_ARG("lighton"))
+        ADD_BOOL(json, "garageLightOn", garage_door.light)
+    if (all || HAS_ARG("motion"))
+        ADD_BOOL(json, "garageMotion", garage_door.motion)
+    if (all || HAS_ARG("obstruction"))
+        ADD_BOOL(json, "garageObstructed", garage_door.obstructed)
 
     // remove the final comma/newline to ensure valid JSON syntax
-    json.erase(json.rfind(",\n"));
-    json.append("\n}"); // close the json
-
+    json[strlen(json) - 2] = 0;
+    // Terminate json with close curly
+    strcat(json, "\n}");
     // Only log if all requested (no arguments).
     // Avoids spaming console log if repeated requests for one value.
     if (all)
-        RINFO("Status requested:\n%s", json.c_str());
+        RINFO("Status requested:\n%s", json);
+    server.sendHeader("Cache-Control", "no-cache, no-store");
+    server.send(200, "application/json", json);
+    return;
+}
+
+void handle_settings()
+{
+    if (!server.authenticateDigest(www_username, www_credentials))
+    {
+        RINFO("In settings request authentication");
+        return server.requestAuthentication(DIGEST_AUTH, www_realm);
+    }
+    return load_page("/settings.html");
+}
 
-    server.send(200, "application/json", json.c_str());
+void handle_logout()
+{
+    RINFO("Handle logout");
+    /*
+    int nHeaders = server.headers();
+    for (int i = 0; i < nHeaders; i++) {
+        RINFO("%s : %s", server.headerName(i).c_str(), server.header(i).c_str());
+    }
+    */
+    return server.requestAuthentication(DIGEST_AUTH, www_realm);
+}
+
+void handle_setgdo()
```

Comment:
I'm not sure we all agree that we want the ability to control the garage door through the web page.